x-pack/plugins/alerting/server/saved_objects/migrations.test.ts

```diff
@@ -571,6 +571,43 @@ describe('7.11.2', () => {
     } as SavedObjectUnsanitizedDoc<RawAlert>;
     expect(isAnyActionSupportIncidents(doc)).toBe(false);
   });
+
+  test('it does not transforms alerts when the right structure connectors is already applied', () => {
```

Comment:
oh I see, what you are saying? Yes I will do that

Reply:
I guess I'm a little curious how well these `incident: {}` actions will survive the Kibana UI experience.  I'm pretty sure on the back-end we'll fail fast because we are always validating these before we use them productively.  But wondering if you edit an alert using one of these actions - what happens?  Scared of seeing some kind of null dereference somewhere, I guess ...

I think that would be the decision point as to whether we need to do _something_ for these in a 7.12.0 migration.  Like add a placeholder `summary` or other required fields in the `incident`, to keep other things from breaking.

Reply:
oh I tried it out to edit an alert when you have this `incident: {}` and we come back to normal. We know it is working because our user who creates the SDH were able to get back to normal after editing all their alert manually.

And yes when you have incident:{} from the SDH, we know the alert get trigger but we never create an action in this connector

Reply:
I don't really understand this bug, but just want to chime in and agree with @pmuellr . It doesn't usually make sense to fix a bug in an already released migration function (unless it's causing data loss) because that buggy code has already run on users's data. A 7.11.2 -> 7.12 upgrade won't get the bug fix applied to the data. So there needs to be a 7.12 migration to guarantee that all the upgrade paths get this bug fix applied.

Reply:
@rudolf we ask user to not migrate from 7.11.0/1 to 7.11.2 because if you do then user will loose the data actions for good. We will have now way to bring it back through migration and you will have to run this [script](https://github.com/elastic/sdh-kibana/issues/1320) to get it back. So we want to stop the bleeding by changing the 7.11.2 migration so user can move forward from 7.11.0/1 to 7.12.x.

Reply:
Ah ok, if this was a data loss bug then yes, all we can do is fix the data loss bug in the 7.11.2 migration, it's too late for users who already upgraded...